### PR TITLE
arm/rp2040: Add I2C driver support

### PR DESCRIPTION
**Pull request by yunkya2**
## Summary
Add I2C driver for Raspberry Pi Pico.

## Impact
rp2040 only

## Testing
New config `raspberrypi-pico:ssd1306` is added to test I2C OLED display (SSD1306).
Graphic samples nx, nxhello, nxdemo, nxlines are working with this config.

### CODE REVIEW

File: boards/arm/rp2040/common/src/rp2040_i2cdev.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ *   Initialize and register i2c driver for the specified i2c port
+ *
+ ****************************************************************************/
+
+int board_i2cdev_initialize(int port)
+{
+  int ret;
+  FAR struct i2c_master_s *i2c;
+
+  _info("Initializing /dev/i2c%d..\n", port);
```
Comment (acassis) [inline boards/arm/rp2040/common/src/rp2040_i2cdev.c:50]: Please change to i2cinfo()

File: boards/arm/rp2040/common/src/rp2040_i2cdev.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  /* Initialize i2c device */
+
+  i2c = rp2040_i2cbus_initialize(port);
+  if (!i2c)
+    {
+      _err("ERROR: Failed to initialize i2c%d.\n", port);
```
Comment (acassis) [inline boards/arm/rp2040/common/src/rp2040_i2cdev.c:57]: please change to i2cerr()

File: boards/arm/rp2040/common/src/rp2040_i2cdev.c [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      return -ENODEV;
+    }
+
+  ret = i2c_register(i2c, port);
+  if (ret < 0)
+    {
+      _err("ERROR: Failed to register i2c%d: %d\n", port, ret);
```
Comment (acassis) [inline boards/arm/rp2040/common/src/rp2040_i2cdev.c:64]: Ditto

File: boards/arm/rp2040/common/src/rp2040_i2cdev.c (hunk 1 above)
Comment (yunkya2) [reply boards/arm/rp2040/common/src/rp2040_i2cdev.c:50]: I got it.

**Comment by acassis** [comment]
Hi @yunkya2 please fix it:
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/arch/arm/src/rp2040/rp2040_i2c.c:515:86: error: Long line found

File: arch/arm/src/rp2040/rp2040_i2c.c [hunk 4]
```diff
@@ -0,0 +1,832 @@
...
+  wd_start(&priv->timeout, I2C_TIMEOUT,
+           rp2040_i2c_timeout, (wdparm_t)priv);
+  i2c_reg_write(priv, RP2040_I2C_IC_DATA_CMD_OFFSET,
+                (uint32_t)msg->buffer[i] |
+                (last ? RP2040_I2C_IC_DATA_CMD_STOP : 0));
+
+  /* Enable TX_EMPTY interrupt for determine transfer done. */
+
+  i2c_reg_rmw(priv, RP2040_I2C_IC_INTR_MASK_OFFSET,
+              RP2040_I2C_IC_INTR_STAT_R_TX_EMPTY, RP2040_I2C_IC_INTR_STAT_R_TX_EMPTY);
```
Comment (acassis) [inline arch/arm/src/rp2040/rp2040_i2c.c:515]: Please break this line to put the last parameter in the new line, this way it will pass in the CI test

Comment (yunkya2) [reply arch/arm/src/rp2040/rp2040_i2c.c:515]: Oops. I overlooked the style error before commit. Thanks.

**Pull request merged**